src/gui/DatabaseWidget.cpp

```diff
@@ -820,25 +798,37 @@ void DatabaseWidget::mergeDatabase(bool accepted)
         merger.merge();
     }
 
-    m_databaseOpenMergeWidget->clearForms();
-    setCurrentWidget(m_mainWidget);
+    switchToMainView();
     emit databaseMerged(m_db);
 }
 
+/**
+ * Unlock the database.
+ *
+ * @param accepted true if the unlock dialog or widget was confirmed with OK
+ */
 void DatabaseWidget::unlockDatabase(bool accepted)
```

Comment:
This function is very bloated and almost all its information is inferred from knowledge of the sender. I would rather this be beefed up and turned public such that the Intent and Database are sent explicitly. No knowledge of any dialogs or widgets should be required.

Reply:
See above.

Reply:
As discussed, we can look for better solutions in future iterations